src/libraries/System.Private.Xml/src/System/Xml/Core/XmlRawWriter.cs

```diff
@@ -218,6 +218,15 @@ public override void WriteNode(System.Xml.XPath.XPathNavigator navigator, bool d
             throw new InvalidOperationException(SR.Xml_InvalidOperation);
         }
 
+        public override void WriteValue(decimal value)
+        {
+            Span<char> destinationSpan = stackalloc char[64];
```

Comment:
how do we know 64 is sufficient?

Reply:
Decimal.MaxValue is 79,228,162,514,264,337,593,543,950,335, which is 29 digits. Add a sign and a decimal point, and it's still well under 64 characters. This works for any scale in the range [0, 28] that Decimal supports.

Anyway, if 64 characters were not enough, then TryFormat would just return false and WriteValue would fall back to the base implementation and succeed, although with some wasted work.

Reply:
use 31 instead and write a comment that's what `Number.FormatDecimal` uses and point to https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Number.NumberBuffer.cs#L13

Note that this is number of characters - for bytes it will be multiplied by 2 assuming UTF-16 is used